frontend/src/components/TeachersPage/TeachersPage.js

```diff
@@ -0,0 +1,199 @@
+        isOpenConfirmDialog,
+        getCurrentSemester,
+        getDefaultSemester,
+        getAllPublicSemesters,
+        sendTeacherSchedule,
+        selectedTeacherCard,
+        deleteTeacher,
+        showAllTeachers,
+        getDisabledTeachers,
+        handleTeacher,
+    } = props;
+    const [term, setTerm] = useState('');
+    const [isDisabled, setIsDisabled] = useState(false);
+    const [selected, setSelected] = useState([]);
+    const [teacherId, setTeacherId] = useState(-1);
+    const [isOpenMultiSelectDialog, setIsOpenMultiSelectDialog] = useState(false);
+    const [selectedSemester, setSelectedSemester] = useState('');
+    const [confirmDialogType, setConfirmDialogType] = useState('');
+
+    useEffect(() => {
+        getCurrentSemester();
+        getDefaultSemester();
+        getDisabledTeachers();
+        showAllTeachers();
+        getAllSemestersItems();
+        getAllDepartmentsService();
+        getAllPublicSemesters();
+        getPublicClassScheduleListService();
+    }, []);
+
+    const visibleItems = search(isDisabled ? disabledTeachers : enabledTeachers, term, [
+        'name',
+        'surname',
+        'patronymic',
+        'department.name',
+    ]);
+
+    const changeDisable = () => {
+        setIsDisabled((prev) => !prev);
+    };
+
+    const teacherSubmit = (values) => {
+        const sendData = { ...values, department };
+
+        handleTeacher(sendData);
+        clearDepartment();
+    };
+
+    const setEnabledDisabled = (currentTeacherId) => {
+        const findTeacher = [...enabledTeachers, ...disabledTeachers].find(
+            (teacherEl) => teacherEl.id === currentTeacherId,
+        );
+
+        const isDisable = dialogTypes.SET_VISIBILITY_ENABLED !== confirmDialogType;
+        handleTeacher({ ...findTeacher, disable: isDisable });
+    };
+
+    const showConfirmDialog = (id, dialogType) => {
+        setTeacherId(id);
+        setConfirmDialogType(dialogType);
+        setOpenConfirmDialog(true);
+    };
+    const acceptConfirmDialog = () => {
+        setOpenConfirmDialog(false);
+        if (confirmDialogType !== dialogTypes.DELETE_CONFIRM) {
+            setEnabledDisabled(teacherId);
+        } else {
+            deleteTeacher(teacherId);
+        }
+    };
+
+    const cancelSelection = () => {
+        setSelected([]);
+        setIsOpenMultiSelectDialog(false);
+    };
+
+    const sendTeachers = () => {
+        setIsOpenMultiSelectDialog(false);
+        const teachersId = selected.map((item) => {
+            return item.id;
+        });
+        const semesterId = selectedSemester === '' && defaultSemester.id;
+        const { language } = i18n;
+        const data = { semesterId, teachersId, language };
+        sendTeacherSchedule(data);
+        setSelected([]);
+    };
+
+    const isChosenSelection = () => {
+        return selected.length !== 0;
+    };
+
+    const parseDefaultSemester = () => {
+        return {
+            id: defaultSemester.id,
```

Comment:
Could you destructuring  defaultSemester?

Reply:
done